server/src/test/java/org/opensearch/gateway/ReplicaShardBatchAllocatorTests.java

```diff
@@ -800,15 +934,75 @@ public TestBatchAllocator addData(
             }
             data.put(
                 node,
+                new TransportNodesListShardStoreMetadataBatch.NodeStoreFilesMetadataBatch(
```

Comment:
How were the batch tests working earlier without this change ? Trying to understand as to why is this change required now ?

Reply:
Earlier Batch data response object was getting constructed as part of fetchData implementation.

I had to make this change because the current fetchData implementation in this test class assumes there is only one primary shard required for the test setup. For new test case on throttling deciders I required a setup with multiple shards hence storing the data in batch format so that fetchData call can return the data cache as it is. This is similar to what we also have in PrimaryShardBatchAllocatorTests.